src/richie/apps/courses/models/course.py

```diff
@@ -660,6 +660,16 @@ class CourseRunSyncMode(models.TextChoices):
     SYNC_TO_PUBLIC = "sync_to_public", _("Synchronization to public page")
 
 
+class CourseRunCatalogVisibility(models.TextChoices):
+    """Course run catalog visibility choices."""
+
+    BOTH = "both", _("both - show in catalog and allow access to about page")
+    ABOUT = "about", _("about - only allow access to about page")
+    NONE = "none", _(
```

Comment:
You should make the field nullable and make use of the `None` value for this.

Reply:
When I change the `NONE = None` the `course_runs_sync` API enforces that the `catalog_visibility` should be `None` instead of `none`.
```
b'{"catalog_visibility":["\\"none\\" is not a valid choice."]}'
```
Then the API won't have a consistent values on the `catalog_visibility`. The supported values would be: `both`, `about` and `None`.
On Open edX Studio Advanced Settings the user inserts directly one of the values: `both`, `about` and `none`. It is not even a combo box.
So, in my opinion would be much simpler if we use the `none` string. 

Alternatively, we could use another naming, for example `hidden`.

Reply:
I think hidden is a much better name indeed. 
none is connoted in Python and should not be used as a choice for a machine string in my opinion.

Reply:
Ok. On Richie we would use `both`, `about` and `hidden` has valid values for the course run catalog visibility, including the `course_runs_sync` API.
The API clients that integrates with Open edX should dial with this difference.

Reply:
OK we can do it in the backend I presume

Reply:
Yes, I can do it on [richie_openedx_sync](https://github.com/fccn/richie-openedx-sync/blob/master/richie_openedx_sync/tasks.py#L49). If `none` then change it to `hidden`.

```python
        "catalog_visibility": 'hidden' if course.catalog_visibility == 'none' else course.catalog_visibility,
```

Reply:
I think it is LMS specific and should be done in the backend, so probably in this method: https://github.com/openfun/richie/blob/master/src/richie/apps/courses/lms/edx.py#L77

Reply:
Maybe don't mention the "about" page as it does not mean anything in richie?
I think we should be very explicit as it is not easy to understand:
- show on the course page and include in search results
- show on the course page and hide in search results
- hide on the course page and in search results

Reply:
> I think it is LMS specific and should be done in the backend, so probably in this method: https://github.com/openfun/richie/blob/master/src/richie/apps/courses/lms/edx.py#L77

When you wrote "backend" I thought that you were talking about the LMS backend and not the richie backend.
I've changed the `clean_course_run_data` edx method implementation to also transform /adapt the data.
So make it work, I also need to change the order of evaluation on the `course_runs_sync` api. 
Previously: 
1. serializer 
2. clean data
Current: 
1. clean + transform data 
2. serializer


> Maybe don't mention the "about" page as it does not mean anything in richie? I think we should be very explicit as it is not easy to understand:
> 
> * show on the course page and include in search results
> * show on the course page and hide from search results
> * hide on the course page and from search results

I think we should show a simpler state name on the combo box. It simplifies the communication between the persons. We can change its state name or even the value of the `CourseRunCatalogVisibility` class.

Current class with reviewed messages:
```python
class CourseRunCatalogVisibility(models.TextChoices):
    """Course run catalog visibility choices."""

    BOTH = "both", _("both - show on the course page and include in search results")
    ABOUT = "about", _("about - show on the course page and hide from search results")
    HIDDEN = "hidden", _("hidden - hide on the course page and from search results")
```

Proposal class (changed from `about` to `course_page`:
```python
class CourseRunCatalogVisibility(models.TextChoices):
    """Course run catalog visibility choices."""

    BOTH = "both", _("both - show on the course page and include in search results")
    COURSE_PAGE = "course_page", _("course_page - show on the course page and hide from search results")
    HIDDEN = "hidden", _("hidden - hide on the course page and from search results")
```
@sampaccoud  What do you think about this proposal ?

Reply:
Nice!
Could go further and change the name BOTH as well? Maybe:  "course_and_search", "course_only" and "hidden"?

Reply:
> Nice! Could go further and change the name BOTH as well? Maybe: "course_and_search", "course_only" and "hidden"?

Yes, I will.

Reply:
Done. 
Also adapt the edx values of the catalog visibility on the method `clean_course_run_data`.